app_legacy/Helpers/render/game.php

```diff
@@ -84,7 +88,33 @@ function renderGameTitle(?string $title = null, bool $tags = true): string
         $updateHtml($html, $matches[0], $tags ? " $span" : '');
     }
 
-    return $html;
+    return "<div>$html</div>";
+}
+
+/**
+ * Render game console with icon of specified size.
+ * Font size is calculated dynamically from icon size.
+ */
+function renderGameConsole(
+    string $consoleName, int $consoleID, int $size, bool $avatar = false
+): string {
+    // Get console icon URL path based on given ID
+    $getConsoleIconSrc = function ($consoleID) {
+        $cleanSystemShortName = Str::lower(str_replace("/", "", config("systems.$consoleID.name_short")));
+        $iconName = Str::kebab($cleanSystemShortName);
+
+        return asset("assets/images/system/$iconName.png");
+    };
+
+    $fallBackConsoleIcon = asset("assets/images/system/unknown.png");
+    $iconSrc = $getConsoleIconSrc($consoleID);
+    $class = $avatar ? ' avatar' : '';
+    $textSize = round(($avatar ? .75 : .6) * $size);
+
+    return "<div class='console$class'>"
+        . "<img src='$iconSrc' width='$size' height='$size' alt='' onerror='this.src=\"$fallBackConsoleIcon\"'>"
+        . "<span style='font-size: {$textSize}px'>$consoleName</span>"
+        . "</div>";
```

Comment:
nit: Heredoc syntax can be used to avoid the string concatenation and quote escapes.

```suggestion
    return <<<HTML
        <div class='console$class'>
            <img src='$iconSrc' width='$size' height='$size' alt='' onerror='this.src="$fallBackConsoleIcon"'>
            <span style='font-size: {$textSize}px'>$consoleName</span>
        </div>
    HTML;
```